### PR TITLE
Document throwing errors with variables

### LINKED ISSUE #1510: Document passing variables along with errors
Camunda Platform `8.2.0-alpha2` will support passing variables along with errors thrown for jobs. We will need to document this feature.

We need to:
- Update the [gRPC API reference](https://docs.camunda.io/docs/next/apis-clients/grpc/#throwerror-rpc)
- Clarify the variable propagation behavior for [Error Events](https://docs.camunda.io/docs/next/components/modeler/bpmn/error-events/) (they are not propagated, i.e. they are always created on the local scope of the catching event, and output mappings can be used as always to control how the variables are merged into the process instance)

We may need to:
- Describe the use case in the [error handling concept](https://docs.camunda.io/docs/next/components/concepts/workflow-patterns/#handling-errors)

Introduced with:
- https://github.com/camunda/zeebe/pull/10882

### PR DESCRIPTION
**Pull request by korthout**
## What is the purpose of the change

_Briefly describe the change you are making, this helps the reviewer by providing an overview._

Documents the new 8.2 ability to pass variables with a thrown error through the `ThrowError` RPC. These variables can be used in the process and may provide additional details about the error.

closes #1510 

## Are there related marketing activities

_Are there any marketing activities related to the related feature or component? If so, please briefly describe them._

Part of the 8.2 release.

## When should this change go live?

_Does this change need to be released before a certain date or milestone? If so, when?_

Along with the 8.2 release.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by christinaausley — APPROVED** [review]
Technical review in latest commit -- this is only for the `next` version of docs, yes?

File: docs/components/modeler/bpmn/error-events/error-events.md [hunk 1]
```diff
@@ -82,6 +82,12 @@ It's much more important to look at how you _react_ to certain errors. Even a te
 
 In general, we recommend talking about business reactions, which are modeled in your process, and technical reactions, which are handled generically using retries or incidents.
 
+## Variable mappings
+
+All error variables are merged into the error catch event. These variables can be merged into the process instance by defining an output mapping at the error catch event.
+
+Visit the documentation regarding [variable mappings](/components/modeler/bpmn/service-tasks/service-tasks.md#variable-mappings) for more information.
```
Comment (remcowesterhoud) [inline docs/components/modeler/bpmn/error-events/error-events.md:89]: ❓ Why do we link to the service task variable mappings here? Wouldn't it make more sense to link to https://docs.camunda.io/docs/next/components/concepts/variables/#inputoutput-variable-mappings?

**Review by remcowesterhoud — APPROVED** [review]
@korthout Some small questions, otherwise all good from my side.

❓ In the issue it says:

> Describe the use case in the [error handling concept](https://docs.camunda.io/docs/next/components/concepts/workflow-patterns/#handling-errors)

I'm fine with leaving this out, but I wanted to raise it in case you missed it.

**Comment by korthout** [comment]
>❓ In the issue it says:
>
>Describe the use case in the [error handling concept](https://docs.camunda.io/docs/next/components/concepts/workflow-patterns/#handling-errors)
>
>I'm fine with leaving this out, but I wanted to raise it in case you missed it.

@remcowesterhoud Yeah I noticed this, but after reading those, I don't really want to make any changes there actually. Thanks for raising it though

**Comment by akeller** [comment]
My docs push got in the way of your merge. Sorry about that! Merging main and rerunning the build 🤞

**Review by akeller — APPROVED** [review]

**Pull request merged**